### ISSUE TITLE
Tabbing beyond the content of a Drop jumps the tab focus to the top of the page

### ISSUE THREAD

**Issue by phinch** [grommet/grommet#5896]
### Expected Behavior 

It's not clear to me what the globally expected behavior should be - but I would generally expect that if I have a DropButton in a page, then tabbing when the drop is open should allow me to tab through content within the drop, and when there is no more content to tab through, tab should either close the drop, or do nothing.

### Actual Behavior

Tabbing beyond the content of a drop seems to cause the tab order to return to the top of the page, rather than doing either of those things. This can have negative impacts for users who are trying to navigate a page using a keyboard - in my team's use case, we have a Select input inside of a form (which uses DropButton under the hood). When the user is using a keyboard to navigate and fill out the form, which is a use case we are required to cover to meet accessibility standards, when the user tabs within the Select options list, they are taken completely out of the form and it is very hard for them to continue navigating via keyboard.

### URL, screen shot, or Codepen exhibiting the issue

I put together some simple examples here - the issue can also be reproduced on the Grommet website examples for DropButton and Select.
https://codesandbox.io/s/grommet-v2-template-forked-j4c80?file=/index.js

### Steps to Reproduce

1. Navigate to the Select input using tab and open the menu by pressing Enter.
2. Press Tab while the menu is open
3. Observe that Tab does not move focus to the next item on the page; additionally observe that other keyboard interactions, like Escape to close the menu, no longer work

### Your Environment

- Grommet version: 2.19.0

**Comment by eatyourgreens** [comment]
We've run into this same problem. We're setting `dropProps` to `trapFocus: false` on the Select, so that you can tab out of it while it is open. Unfortunately, we then find that tabbing out of the open menu leaves the menu open and sends focus to the first focusable element on the page:
https://github.com/zooniverse/front-end-monorepo/pull/2838#pullrequestreview-883406277

We've looked at tab behaviour for HTML `<select>` elements in different browsers:
- Chrome and Safari (and presumably Edge): An open menu traps focus and Tab is ignored. I think this it what Grommet's `Select` is emulating with `trapFocus: true`.
- Firefox: Tabbing out of an open menu selects the focussed option and sends keyboard focus to the next focusable element in the page (usually the next form input.) We'd really like to be able to do this with the Grommet `Select` too.

**Comment by phinch** [comment]
Hi all (not sure if bumping works here but giving it a try) - does this seem like a valid issue that is worth looking into? Our team is loving the accessibility updates that have gone into the Select input recently, and it seems like this is the last outstanding issue that can make it a little wonky (and mis-matching with the select spec) to use on keyboard

**Comment by jcfilben** [comment]
Hi @phinch thanks for bumping this! It helps us to prioritize and know what issues have interest from the community. This issue is on our list to look into

**Comment by phinch** [comment]
hi @ShimiSun, thanks for the detailed response! You're totally right that this is more of a nice-to-have when it comes to accessibility, rather than an explicit piece of the wcag guidelines. I appreciate you all taking the time to look into it nonetheless :) 

From my team's perspective, swallowing the Tab keyboard action while the menu is open sounds perfect! As I understand it, that would fix the accessibility issues our team had initially reported. The enhancement sounds interesting - I don't think our team would be opposed to it, but it definitely is not something that, in our eyes, would be a must-have for the Select. We saw the experience of being taken to the end of the webpage as the critical piece of behavior here, so as long as that part is resolved, we're 100% happy.

For navigating outside of the drop, I also think it makes sense that the user should not be able to navigate outside of the Select dropdown when it's open to mimic the native behavior. I might hope that swallowing tab events would take care of this? Although from our team's perspective, I don't think we have a strong opinion other than to try to match the wcag standards. 

Finally, as for cases outside of the Select input, the described behavior of the DropButton that I put in the original example feels a bit strange to me, since once you tab out of the Drop you still get taken to the end of the page with no ability to close the Drop - but it definitely doesn't rise to the level of accessibility concern that the Select case does, and I understand that it's not feasible to solve this problem for all Drops because not all Drops should behave the same way. Having the fix applied to the Select input would be a great improvement 😄 Thank you again for looking into it!

**Comment by ShimiSun** [comment]
>Finally, as for cases outside of the Select input, the described behavior of the DropButton that I put in the original example feels a bit strange to me, since once you tab out of the Drop you still get taken to the end of the page with no ability to close the Drop - but it definitely doesn't rise to the level of accessibility concern that the Select case does, and I understand that it's not feasible to solve this problem for all Drops because not all Drops should behave the same way. Having the fix applied to the Select input would be a great improvement 😄 Thank you again for looking into it!

That's a good point, and I think that every component or use case will have its own type of accessibility handling. I'm glad we covered and agreed on the one related to Select. As for the DropButton example you provided, I think that the Drop `inline` prop suggested above will solve that issue since `inline` should position the Drop inline to its owning element (on the DOM) which will resolve the tabbing order from the Drop further.

**Comment by eatyourgreens** [comment]
There's no standard browser behaviour for tabbing out of a select menu, but I think adding `onTab={onSelectOption}` to the select container here would reproduce default Select behaviour in Firefox, without breaking the Grommet Select in Chrome.
https://github.com/zooniverse/front-end-monorepo/issues/3024

**Comment by ShimiSun** [comment]
Thanks for the input @eatyourgreens (what a great alias, I need to introduce you to my son :) ).
If I understand your comment correctly, the behavior onTab that you are suggesting will close the Select's Drop, how do you think that will be backward compatible?

**Comment by eatyourgreens** [comment]
Firefox selects the focused option and closes an open select on tab, so I think that’s backwards compatible. It is a change in behaviour for other browsers, though. 

We’ve had complaints from Firefox users, saying that the tab key is broken for Grommet selects in our app. They’re expecting to be able to select an option with the cursor keys, then close the select with tab. 

It’s annoying that browsers don’t agree on a standard behaviour here. It seems to me that Grommet is following the Chrome behaviour, which is absolutely fine but annoying for Firefox users.

**Comment by eatyourgreens** [comment]
The WAI-ARIA design pattern for combo boxes doesn’t mention any expected behaviour for Tab, so I guess that’s left up to implementers to do whatever they think is best for users. 
https://www.w3.org/TR/wai-aria-practices-1.1/examples/combobox/aria1.1pattern/listbox-combo.html

That WAI-ARIA example does show the roles and states for a combo box. I’m not sure if the Grommet Select uses all of those eg. marking up the list of buttons as a `listbox` and each button as an `option`. Those roles and states might be useful for screen readers.

**Comment by eatyourgreens** [comment]
Apologies, I was testing with an old version of Grommet and I can see that the latest version does use the `listbox` role. The `combobox` role could be useful for announcing the options on text inputs with suggestions. 

This WAI-ARIA 1.2 combo box example has an event handler for Tab, which selects the active option and closes the listbox. 
https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html

It could be that expected Tab behaviour varies between a standard select and a combo box with text input and options.

**Comment by jcfilben** [comment]
Resolved by #7324
